GitUI/GitUICommands.cs

```diff
@@ -670,19 +670,18 @@ public bool StartFixupCommitDialog(IWin32Window owner, GitRevision revision)
 
         public bool StartCommitDialog(IWin32Window owner, bool showOnlyWhenChanges)
         {
+            var form = new FormCommit(this);
             Func<bool> action = () =>
             {
-                using (var form = new FormCommit(this))
-                {
-                    if (showOnlyWhenChanges)
-                        form.ShowDialogWhenChanges(owner);
-                    else
-                        form.ShowDialog(owner);
-                }
+                if (showOnlyWhenChanges)
+                    form.ShowDialogWhenChanges(owner);
+                else
+                    form.ShowDialog(owner);
+                form.Dispose();
                 return true;
             };
 
-            return DoActionOnRepo(owner, true, false, PreCommit, PostCommit, action);
+            return DoActionOnRepo(form, true, false, PreCommit, PostCommit, action);
```

Comment:
What about overloading `DoActionOnRepo` method to support custom pre- and post-event handler types?
Current owner of type `IWin32Window` is provided as parent for modal dialogs, casting it to the specific class and calling it's methods looks a little hacky.

More graceful way is to change `PreCommit` event handler type and give to subscribers special event args class (derived from `GitUIEventArgs`) with features specific for that event.